src/main/scala/com/amazon/deequ/analyzers/QuantileNonSample.scala

```diff
@@ -0,0 +1,307 @@
+package com.amazon.deequ.analyzers
+
+import scala.collection.immutable.ListMap
+import scala.collection.mutable.ArrayBuffer
+import scala.reflect.ClassTag
+import scala.util.control.Breaks._
+
+
+class QuantileNonSample[T](val sketchSize: Int,
+                           val shrinkingFactor: Double = 0.64)
+                          (implicit ordering: Ordering[T],
+                           ct: ClassTag[T]) extends Serializable{
+
+  private var curNumOfCompactors = 0
+  // number of items in compactors
+  private var compactorActualSize = 0
+  // overall capacity of compactors
+  private var compactorTotalSize = 0
+  // initialize with ArrayBuffer, add compactors later
+  private var compactors = ArrayBuffer[NonSampleCompactor[T]]()
+  expand()
+
+  // get and set method for the serialize/deserialize functions
+  def getShrinkingFactor: Double = {
```

Comment:
Why do we need the accessors here? We can directly access the variables, it does not make a difference

Reply:
Actually what I was thinking is that we should set the variables to private (except sketchSize) to be more secure. If the get/set accessors are public, then there's no difference between setting variables to public and remove accessors. However, maybe later we could add stricter access modifier to these accessors. In that case it would make a difference. But this is only my point so feel free to give your advice on this.

Reply:
You can do that if you feel that it is necessary, but IMHO, this Java Bean pattern is rarely used in scala.

Reply:
After a second thought and discuss with Arun, we decide to remove the get/set methods! Thank you for your great advice!